src/starlite_saqlalchemy/lifespan.py

```diff
@@ -41,7 +48,7 @@ async def _redis_ready() -> None:
 
 async def before_startup_handler(_: starlite.Starlite) -> None:
     """Do things before the app starts up."""
-    if settings.app.CHECK_DB_READY:
+    if constants.IS_SQLALCHEMY_INSTALLED and settings.app.CHECK_DB_READY:
         await _db_ready()
-    if settings.app.CHECK_REDIS_READY:
+    if constants.IS_REDIS_INSTALLED and settings.app.CHECK_REDIS_READY:
```

Comment:
Actually, there already is two separate ones, made this so that they always get called, but only do anything if their required dependency is available.